plugins/pulp_python/plugins/importers/importer.py

```diff
@@ -183,7 +185,11 @@ def upload_unit(self, repo, type_id, unit_key, metadata, file_path, conduit, con
         :rtype:           dict
         """
         package = models.Package.from_archive(file_path)
-        package.save_and_import_content(file_path)
+        try:
+            package.save_and_import_content(file_path)
+        except NotUniqueError:
+            package = package.__class__.objects.filter(**package.unit_key).first()
```

Comment:
What about the uploaded file? Where will it go? Will it get garbage collected by the task working dir mechanism? I'll assume yes, just want to make sure.